phoenix-core/src/it/java/org/apache/phoenix/end2end/LogicalTableNameExtendedIT.java

```diff
@@ -0,0 +1,281 @@
+                        put.addColumn(QueryConstants.DEFAULT_COLUMN_FAMILY_BYTES, Bytes.toBytes("V2"),
+                                PInteger.INSTANCE.toBytes(32));
+                        put.addColumn(QueryConstants.DEFAULT_COLUMN_FAMILY_BYTES, Bytes.toBytes("V3"),
+                                PInteger.INSTANCE.toBytes(33));
+                        htable.put(put);
+                    }
+
+                    IndexToolIT.runIndexTool(true, false, schemaName, tableName, indexName);
+                    rs = conn.createStatement().executeQuery("SELECT * FROM " +  fullIndexName + " WHERE \":PK1\"='PK30'");
+                    assertEquals(true, rs.next());
+
+                    // Drop tables
+                    conn2.createStatement().execute("DROP TABLE " + fullTableName);
+                    // check that the physical data table is dropped
+                    assertEquals(false, admin.tableExists(TableName.valueOf(fullNewTableHName)));
+
+                    // check that index is dropped
+                    assertEquals(false, admin.tableExists(TableName.valueOf((schemaName + ":NEW_IDXTBL_" + indexName))));
+                }
+            }
+        }
+    }
+
+    @Test
+    public void testUpdatePhysicalTableName_alterTable() throws Exception {
+        String schemaName = "S_" + generateUniqueName();
+        String tableName = "TBL_" + generateUniqueName();
+        String indexName = "IDX_" + generateUniqueName();
+        String fullTableName = SchemaUtil.getTableName(schemaName, tableName);
+        String fullTableHName = schemaName + ":" + tableName;
+        String fullIndexName = SchemaUtil.getTableName(schemaName, indexName);
+        String fullIndexHName = schemaName + ":" + indexName;
+        String fullNewTableHName = schemaName + ":NEW_TBL_" + tableName;
+        try (Connection conn = getConnection(propsNamespace)) {
+            try (Connection conn2 = getConnection(propsNamespace)) {
+                test_bothTableAndIndexHaveDifferentNames(conn, conn2, schemaName, tableName, indexName);
+                conn2.setAutoCommit(true);
+
+                conn2.createStatement().execute("ALTER TABLE " + fullTableName + " ADD new_column_1 VARCHAR(64) CASCADE INDEX ALL");
+                conn2.createStatement().execute("UPSERT INTO " + fullTableName + " (PK1, V1, new_column_1) VALUES ('a', 'v1', 'new_col_val')");
+                ResultSet
+                        rs =
+                        conn2.createStatement().executeQuery("SELECT \"0:NEW_COLUMN_1\" FROM " + fullIndexName);
+                assertEquals(true, rs.next());
+                rs = conn.createStatement().executeQuery("SELECT NEW_COLUMN_1 FROM " + fullTableName + " WHERE NEW_COLUMN_1 IS NOT NULL");
+                assertEquals(true, rs.next());
+                assertEquals(false, rs.next());
+
+                // Drop row and check
+                conn.createStatement().execute("ALTER TABLE " + fullTableName + " DROP COLUMN NEW_COLUMN_1");
+
```

Comment:
Looks like we drop but don't check as the comment says we will? Or is the lack of an exception the only check we need here?

Reply:
Will update the comment. Yes lack of exception